# STAT 301: Final Review

In [1]:
# The code will not be displayed in the quiz
library(tidyverse, quietly = T)
library(broom, quietly = T)
library(mltools)
library(leaps)
library(glmnet)

library(ISLR)
dat <- College

── Attaching packages ─────────────────────────────────────── tidyverse 1.3.2 ──
✔ ggplot2 3.3.6      ✔ purrr   0.3.5 
✔ tibble  3.1.8      ✔ dplyr   1.0.10
✔ tidyr   1.2.1      ✔ stringr 1.4.1 
✔ readr   2.1.3      ✔ forcats 0.5.2 
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()

Attaching package: ‘mltools’


The following object is masked from ‘package:tidyr’:

    replace_na


Loading required package: Matrix


Attaching package: ‘Matrix’


The following objects are masked from ‘package:tidyr’:

    expand, pack, unpack


Loaded glmnet 4.1-2



## Question 1

A LASSO model fit with `lambda.min` has less input variables than the LASSO model fit with lambda.1se

- T or F


## Question 2

We want to build a model to predict whether or not students would pass a given undergraduate course based on grades of pre-requisite courses, general GPA, and other variables related to study habits. Which of the following models would you recommend:

- a Logistic regression

- a Poisson regression

- a Linear regression


## Question 3

Related to the previous problem, a LASSO regularized method is used to estimate the model while selecting the most important variables to predict. LASSO will select *all* variables that are statistically significant at 1% level. 

- T or F


## Question 4

We want to build an additive multiple linear regression (MLR) to understand the relation between the weigth of newborns and the length of the pregnancy (gestation period, days). We know that many demographic variables should be considered, including for example the smoking and drinking habits of the mother. Since many variables are categorical with more than 2 levels, we can't use many selection packages available in R. 

You plan to write code to correctly implement a forward selection algorithm to select variables for your analysis. Which measure can you use to choose the next variable to be added in the model?

- an F-statistic

- the RSS 

- the cross-validation test MSE

- none of the above

## Question 5

Suppose we want to study the effect of smoking on the risk of getting a heart disease. A logistic regression is estimated using the following variables:

`disease`: with values 0 if no and 1 if yes

`smoking`: a categorical variable with levels `Y` if the patient smokes, `P` if the patient did smoke and quit, and `N` for never smoked

The summary table obtained by `tidy` and argument `exponentiate = TRUE` shows an estimate of 1.46 for the term `smokingP` and 1.78 for `smokingY`. The corresponding p-values are below 0.05.

Based on this results we conclude that:

- (A) The current smoking group is 0.78 times more likely to have a heart disease than not to, compared to the non-smoking group.

- (B) The current smoking group is 1.78 times more likely to have a heart disease than not to, compared to the non-smoking group.

- (C) The current smoking group is 0.78 times more likely to have a heart disease than not to, compared to the former-smokers group.

## Question 6

Suppose we want to study the effect of smoking on the risk of getting a heart disease. A logistic regression is estimated using the following variables:

`disease`: with values 0 if no and 1 if yes

`smoking`: a categorical variable with levels `Y` if the patient smokes, `P` if the patient did smoke and quit, and `N` for never smoked

`weight`: a continuous variable


| Coefficient | estimate|p.value|
|------|------|------|
|   Intercept  | -1.93|<0.001|
|------|------|------|
smokingY| 0.58|<0.001|
|------|------|------|
smokingP| 0.38|<0.001|
|------|------|------|
weight| 0.21|<0.001|

Which of the following codes was used to fit the model summarized in the table:

- `lm(weight ~ smoking)`

- `lm(disease ~ smoking)`

- `glm(disease ~ smoking)`

- `glm(disease ~ smoking * weight)`

- `glm(disease ~ smoking + weight)`

- `glm(disease ~ smokingY + weight)`

## Question 7

You want to investigate factors that affect the number of applications received (`Apps`) by universities. Using the dataset `College` from ISLR, you study the relation of this response variable with the number of accepted applications (`Accept`, continuous variable) and the type of University (`Private`, categorical variable with 2 levels).  

You fit an linear regression using `lm()` and obtain the following output results using the functions `tidy()` and `glance`, respectively:

In [2]:
# The code will not be displayed in the quiz

lm_college_int <- lm(Apps ~ Accept * Private, data = dat)
tidy(lm_college_int)

glance(lm_college_int)

term              estimate    std.error    statistic  p.value      
1 (Intercept)       -28.7028501 132.40598581 -0.2167791  8.284377e-01
2 Accept              1.4693034   0.02529710 58.0818971 3.544189e-284
3 PrivateYes        -72.8622683 151.82962279 -0.4798949  6.314379e-01
4 Accept:PrivateYes   0.1233214   0.04672575  2.6392589  8.476124e-03

r.squared adj.r.squared sigma    statistic p.value df logLik    AIC     
1 0.8914199 0.8909985     1277.762 2115.389  0       3  -6658.286 13326.57
  BIC      deviance   df.residual nobs
1 13349.85 1262057689 773         777

**Question 7.1**

You believe that this model is too complex and that the number of applications received (Apps) by universities depends *only* on the number of accepted applications (Accept). Which of the following results can be used to decide:

- (A) the `p.value` given by `tidy` and equal to 3.544189e-284

- (B) the `p.value` given by `tidy` and equal to 8.476124e-03

- (C) the `p.value` given by `glance` and equal to 0

- (D) the `r.squared` given by `glance` and equal to 0.8914199

- (E) the `adj.r.squared` given by `glance` and equal to 0.8909985

- (F) none of the above, we need the `r.squared` of `lm(Apps ~ Accept, data=College)` to compare with 0.8914199

- (G) none of the above, we need the `adj.r.squared` of `lm(Apps ~ Accept, data=College)` to compare with 0.8909985

- (H) none of the above


**Question 7.2**

We want to compare the additive model `lm(Apps ~ Accept + Private, data=College)` with the given model. Which of the following results can you use: 

- (A) the `p.value` given by `tidy` and equal to 3.544189e-284

- (B) the `p.value` given by `tidy` and equal to 8.476124e-03

- (C) the `p.value` given by `glance` and equal to 0

- (D) the `r.squared` given by `glance` and equal to 0.8914199

- (E) the `adj.r.squared` given by `glance` and equal to 0.8909985

- (F) none of the above, we need the `r.squared` of `lm(Apps ~ Accept, data=College)` to compare with 0.8914199

- (G) none of the above, we need the `adj.r.squared` of `lm(Apps ~ Accept, data=College)` to compare with 0.8909985

- (H) none of the above

## Question 8

A hospital is planning to investigate whether a new exercise routine can reduce the recover time of patients who underwent a knee surgery. A group of researchers will organize a randomized experiment assigning the new routine to 100 of the patients and the current routine to other 100 patients.  

The hospital decides to check the results every day after the patients finish their daily routine. 

The researchers want to examine the results of the interim tests after using an adjustment but they are unsure if using a Bonferroni's method will be more beneficial than a Pocock's method. Which of the following claims is correct:

- (A) While both methods will control the type I error rate, Bonferroni will result in a greater loss of power since the test will be very conservative when conducting 30 interim tests. I recommend Pocock's method

- (B) I recommend Pocock's because Bonferroni's method is not suitable for dependent tests

- (C) I recommend Bonferroni's because Pocock's method gives us the same critical value for each experiment

- (D) I recommend the Bonferroni's method because it is more conservative, we can adjust the significance level to control the type I error.